plugins/Kaleidoscope-DynamicMacros/src/kaleidoscope/plugin/DynamicMacros.cpp

```diff
@@ -136,6 +134,11 @@ void DynamicMacros::play(uint8_t macro_id) {
   uint16_t pos;
   Key key;
 
+  // If the requested ID is higher than the number of macros we found during the
+  // cache update, bail out. Our map beyond `macro_count_` is unreliable.
+  if (macro_id >= macro_count_)
+    return;
+
   pos = storage_base_ + map_[macro_id];
 
   while (true) {
```

Comment:
Mhm. But those kind of problems - sending less data than what the firmware expects - is fairly common among the plugins, because guarding against that is too costly.

Reply:
I don't think that's "sending less data".  I mean when a macro is being written, and it doesn't fit in the storage slice, but the last byte is `MACRO_ACTION_STEP_KEYDOWN`, which causes two additional bytes to get written—past the end of the slice—before the `while` condition terminates the loop.